src/main/java/de/tum/in/www1/artemis/repository/ApollonDiagramRepository.java

```diff
@@ -12,4 +15,6 @@
 @Repository
 public interface ApollonDiagramRepository extends JpaRepository<ApollonDiagram, Long> {
 
+    @Query("SELECT a FROM ApollonDiagram a WHERE a.courseId = :#{#courseId} OR a.courseId is null")
```

Comment:
I think the native query is not necessary here. `List<ApollonDiagram> findDiagramsByCourse(Long courseId);` should be enough and the JPA implementation will do all the magic.

E.g. compare to implementation in `TutorLeaderboardComplaintsViewRepository.java`.